src/packet.rs

```diff
@@ -216,13 +207,67 @@ fn packet(i: &[u8]) -> nom::IResult<&[u8], (u8, Packet<'_>)> {
                 pkt.extend(last.1);
                 pkt
             } else {
-                Packet(last.1, Vec::new())
+                Packet(Vec::from(last.1))
             };
             (seq, pkt)
         },
     )(i)
 }
 
+pub struct SwitchableConn<T: Read + Write>(Option<EitherConn<T>>);
```

Comment:
I don't think this needs to be `pub`? Or am I missing something? It can probably just be `pub(crate)`.